example/newreposecret/main.go

```diff
@@ -0,0 +1,197 @@
+	if token == "" {
+		return nil, nil, errors.New(`no API_GITHUB_TOKEN was found in the environment variables.
+This is needed to authenticate with github. Please generate a github token that has access to do what you are trying to do
+and export it as an environment variable.
+
+ex:
+	export API_GITHUB_TOKEN="<token contents string>"
+`)
+	}
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+	return ctx, client, nil
+}
+
+// AddRepoSecret will add a secret value to a given github repo for a given owner
+// It encrypts the secret using sodium before sending the secret to github api
+// therefore requires libsodium to be installed on the machine running this code
+// https://formulae.brew.sh/formula/libsodium
+// Github is very picky over formats things need to be to be sent, and not very descriptive in how
+// they will be sent to the user
+// To get a secret uploaded you need to get the public key of the repo that will be receiving the
+// secret. This key is used to encrypt the secret before transport on the sending side, and then decrypted by github.
+// Once you have the public key you need to encrypt the secret with sodiumlib before sending it.
+// The public key comes base64 encoded, and sodiumlib expects it to not be base64 encoded so you need
+// to decode the public key before using it.
+// Once the public key is decoded you need to convert the string secret into bytes.
+// once you have the public key decoded, and the secret string in bytes you can encrypt it using
+// sodium.CryptoBoxSeal
+// That will produce the correctly encrypted secret as bytes, but you need to then convert it to
+// a base64 encoded string. After doing that you can use that base64 encoded string as the encrypted value
+// to be part of the github.EncodedSecret type.
+// The name is the string (no encoding, or base64 needed) of the secret name that will appear in github secrets
+// then the KeyID will be the public key of the repo's ID, which is gettable from the public key's GetKeyID method.
+// Finally you can pass that object in and have it be created or updated in github.
+func AddRepoSecret(ctx context.Context, client *github.Client, owner string, repo string, secretName string, secretValue string) (string, error) {
+	publicKey, _, err := client.Actions.GetRepoPublicKey(ctx, owner, repo)
+	if err != nil {
+		return "", err
+	}
+
+	encryptedSecret, err := encryptSecretWithPublicKey(publicKey, secretName, secretValue)
+	if err != nil {
+		return "", err
+	}
+
+	_, err = client.Actions.CreateOrUpdateRepoSecret(ctx, owner, repo, encryptedSecret)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Actions.CreateOrUpdateRepoSecret returned error: %v", err))
+	}
+
+	return secretName, nil
+}
+
+func encryptSecretWithPublicKey(publicKey *github.PublicKey, secretName string, secretValue string) (*github.EncryptedSecret, error) {
```

Comment:
```suggestion
func encryptSecretWithPublicKey(publicKey *github.PublicKey, secretName, secretValue string) (*github.EncryptedSecret, error) {
```

Reply:
Is this intended to remove the typing of the secret name parameter? that is the only change I can see here and I just wanted to make sure that is what you intended. I don't off top of my head understand why we would want to do that, but am open to reasons why!

Reply:
Ah, sorry for dropping this suggestion without explaining it...

In Go, if two or more successive arguments to a method all have the same type, you can combine them together (which is the recommended idiomatic Go style), so:

```
func a(b int, c int, d int, e int, h string, i string j string) {}
```

becomes:

```
func a(b, c, d, e int, h, i, j string) {}
```

Reply:
Ahh thank you so much for the explanation! I thought it was a mistake or something, but now I learned something new! Updated!